modules/templates/vars/vars.go

```diff
@@ -0,0 +1,101 @@
+// Copyright 2022 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package vars
+
+import (
+	"fmt"
+	"strings"
+)
+
+// ErrWrongSyntax represents a wrong syntax with a tempate
+type ErrWrongSyntax struct {
+	Template string
```

Comment:
I think it might be beneficial to serve a reason as well in this case, i.e. `Reason: "\"{\" is not allowed inside another match"`:
```suggestion
	Template string
	Reason string
```